### PR TITLE
Use keyPressEvent instead of eventFilter to handle keyboard events

### PR DESCRIPTION
**Pull request by ccordoba12**
- This solves problems with PyQt5 and helps to simplify things.

### CODE REVIEW

File: qtconsole/completion_widget.py [hunk 1]
```diff
@@ -85,16 +56,24 @@ def eventFilter(self, obj, event):
                 else:
                     self.cancel_completion()
                     if target:
-                        target.event(event)
+                        try:
+                            target.event(event)
+                        except RuntimeError:
+                            pass
```
Comment (vidartf) [inline qtconsole/completion_widget.py:62]: Why is the `try ... except` necessary?

**Comment by vidartf** [comment]
Everything seems to be working, so I'll just merge, but I'm still interested in the reason for the `try ... except` block.

File: qtconsole/completion_widget.py (hunk 1 above)
Comment (ccordoba12) [inline qtconsole/completion_widget.py:62]: I was getting a `RuntimError` when moving focus out of the completion with a mouse click on the main qtconsole window.

Comment (vidartf) [inline qtconsole/completion_widget.py:62]: Does things work for you without this block (`if target: ...`)?

Comment (ccordoba12) [inline qtconsole/completion_widget.py:62]: Yes, things work correctly without that block. If they also work for you, you should remove it :-)

Comment (vidartf) [inline qtconsole/completion_widget.py:62]: Did you keep or remove the `return True` statement?

**Pull request merged**